docs/cheaha/slurm/slurm_tutorial.md

```diff
@@ -0,0 +1,528 @@
+
+```py linenums="1"
+#!/bin/bash
+#SBATCH --job-name=multijob
+#SBATCH --nodes=1
+#SBATCH --ntasks=3
+#SBATCH --cpus-per-task=1
+#SBATCH --mem=1G
+#SBATCH --partition=express
+#SBATCH --time=01:00:00
+#SBATCH --output=%x_%j.out
+#SBATCH --error=%x_%j.err
+
+### Loading Anaconda3 module to activate `pytools-env` conda environment
+module load Anaconda3
+conda activate pytools-env
+
+### Runs the script `python_test.py` in parallel with distinct inputs and ensures synchronization
+srun --nodes=1 --ntasks=1 python python_script_new.py 1 100000 &
+srun --nodes=1 --ntasks=1 python python_script_new.py 100001 200000 &
+srun --nodes=1 --ntasks=1 python python_script_new.py 200001 300000 &
+wait
+```
+
+Copy the following python script and call it as `python_script_new.py`. The input file takes two command-line arguments i.e. the `start` and `end` values. The script uses these values to creates an array and compute the sum of its elements using numpy. The above batch script runs three parallel instances of this Python script with different inputs.
+
+```py linenums="1"
+import sys
+import numpy as np
+
+### Verify if correct number of command-line arguments is provided
+if len(sys.argv) != 3:
+  print("Usage: python python_script_new.py <start> <end>")
+  sys.exit(1)
+
+### Passing start and end values from command-line arguments
+start = int(sys.argv[1])
+end = int(sys.argv[2])
+
+### Create an array from start to end using numpy
+input_array = np.arange(start, end)
+
+### Perform addition on the array elemnts using numpy's sum function
+sum_result = np.sum(input_array)
+
+### Print Input Range and Sum
+print("Input Range: {} to {}, Sum: {}".format(start, end, sum_result))
+```
+
+The below ouptut shows that each line corresponds to the output of one parallel execution of python script with specific input ranges.
+
+```bash
+$cat multijob_27099591.out
+
+Input Range: 1 to 100000, Sum: 4999950000
+Input Range: 200001 to 300000, Sum: 24999750000
+Input Range: 100001 to 200000, Sum: 14999850000
+```
+
+The `sacct` report indicates that three CPUs have been allocated. The python script executes with unique task IDs 27099591.0,27099591.1,27099591.2.
+
+```bash
+$ sacct -j 27099591
+
+       JobID    JobName  Partition    Account  AllocCPUS      State ExitCode 
+------------ ---------- ---------- ---------- ---------- ---------- -------- 
+27099591       multijob    express      USER          3  COMPLETED      0:0 
+27099591.ba+      batch                 USER          3  COMPLETED      0:0 
+27099591.ex+     extern                 USER          3  COMPLETED      0:0 
+27099591.0       python                 USER          1  COMPLETED      0:0 
+27099591.1       python                 USER          1  COMPLETED      0:0 
+27099591.2       python                 USER          1  COMPLETED      0:0 
+```
+
+## Example 4: Array Job
```

Comment:
I think it would be helpful to have one more example:

`## Example : Array Job - Files`

Please add a separate issue for this suggestion.

Reply:
Created an issue for now.

Reply:
Thanks!